Print newline after doc comments before attributes

zploskey opened the pull request:
Forces a line break following doc comments before the first non-comment attribute on a BuckleScript external.

**Before:**

```reason
/** doc comment */ [@bs.send] external url : t => string = "";

/**
 * Short multiline doc comment
 */ [@bs.send]
external url : t => string = "";

/** Longer doc comment before an attribute on an external. */ [@bs.send]
external url : t => string = "";

/* normal comment */
[@bs.send] external url : t => string = "";
```

**After:**

```reason
/** doc comment */
[@bs.send]
external url : t => string = "";

/**
 * Short multiline doc comment
 */
[@bs.send]
external url : t => string = "";

/** Longer doc comment before an attribute on an external. */
[@bs.send]
external url : t => string = "";

/* normal comment */
[@bs.send] external url : t => string = "";
```

Fixes #1857.

zploskey commented:
Rebased and added an additional test case for short multi-line doc comments preceding externals.

chenglou commented:
cc @rickyvetter

jordwalke commented:
Are there sufficient test cases on non-extern items that have doc comments? (Like type definitions and regular let bindings?) If not, would you care to add them so I can merge?

zploskey commented:
I rebased this on master and reworked it to use `partitionAttributes` now that it can handle doc comments again. If there is a simpler way to do something here please let me know. I'm eager to get this merged so I don't have to keep padding out my doc comments just so the attributes don't wind up on the same line as them.

zploskey commented:
CC @IwanKaramazow

IwanKaramazow commented:
@zploskey This looks great! Thanks for the effort, I'll ping @chenglou to make sure it gets merged.

chenglou commented:
Good stuff. Thanks @zploskey!
